e2e/indexer.e2e-spec.ts

```diff
@@ -21,46 +21,64 @@ describe('Indexer', () => {
-        const taprootOutput = walletHelper.generateAddresses(1, 'p2tr')[0];
-        const p2wkhOutputs = walletHelper.generateAddresses(6, 'p2wpkh');
-        const utxos: UTXO[] = [];
+    const addressTypes: AddressType[] = [
+        AddressType.P2WPKH,
+        AddressType.P2TR,
+        AddressType.P2PKH,
+        AddressType.P2SH_P2WPKH,
+    ];
 
-        for (const output of p2wkhOutputs) {
-            const utxo = await walletHelper.addFundToUTXO(output, 1);
-            utxos.push(utxo);
-        }
+    it.each(addressTypes)(
+        '%s - should ensure that the correct silent block is fetched',
+        async (addressType) => {
+            const taprootOutput = walletHelper.generateAddresses(
+                1,
+                AddressType.P2TR,
+            )[0];
+            const outputs = walletHelper.generateAddresses(6, addressType);
+            const utxos: UTXO[] = [];
 
-        const { transaction, txid, blockhash } =
-            await walletHelper.craftAndSendTransaction(
-                utxos,
-                taprootOutput,
-                5.999,
-                0.001,
-            );
+            for (const [index, output] of outputs.entries()) {
+                const utxo = await walletHelper.addFundToUTXO(
+                    output,
+                    1,
+                    addressType,
+                    index,
+                );
+                utxos.push(utxo);
+            }
 
-        const blockCount = await walletHelper.getBlockCount();
-        const transformedTransaction = transactionToEntity(
-            transaction,
-            txid,
-            blockhash,
-            blockCount,
-            p2wkhOutputs,
-        );
+            const { transaction, txid, blockhash } =
+                await walletHelper.craftAndSendTransaction(
+                    utxos,
+                    taprootOutput,
+                    5.999,
+                    0.001,
+                );
 
-        const silentBlock = new SilentBlocksService(
-            {} as any,
-            {} as any,
-        ).encodeSilentBlock([transformedTransaction]);
+            const blockCount = await walletHelper.getBlockCount();
+            const transformedTransaction = transactionToEntity(
+                transaction,
+                txid,
+                blockhash,
+                blockCount,
+                outputs,
+            );
 
-        await new Promise((resolve) => setTimeout(resolve, 15000));
-        const response = await apiHelper.get(
-            `/silent-block/hash/${blockhash}`,
-            {
-                responseType: 'arraybuffer',
-            },
-        );
+            const silentBlock = new SilentBlocksService(
+                {} as any,
+                {} as any,
+            ).encodeSilentBlock([transformedTransaction]);
 
-        expect(response.data).toEqual(silentBlock);
-    });
+            await new Promise((resolve) => setTimeout(resolve, 15000));
```

Comment:
No action required for this PR
but we should be able to get rid of this once we start listening for WS updates